core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/handler/BaseClientHandler.java

```diff
@@ -77,6 +78,13 @@ static <InputT extends SdkRequest, OutputT> InterceptorContext finalizeSdkHttpFu
 
         request = modifyEndpointHostIfNeeded(request, clientConfiguration, executionParams);
 
+        try {
+            URI endpoint = new URI(request.protocol(), request.host(), null, null);
+            executionContext.metricCollector().reportMetric(CoreMetric.SERVICE_ENDPOINT, endpoint);
+        } catch (URISyntaxException e) {
+            throw new IllegalArgumentException(e);
+        }
```

Comment:
Catching and rethrowing exception to not change method signature.
`URI.create` does not throw exception, but requires parsing a string.